$NetBSD: patch-ai,v 1.1 2008/10/15 00:56:45 bjs Exp $

69dadf231283fadcb117b4d9e799e8633a0e4dab

force alignment of arg pointer in potential SSE users

Fix for bug 17824

--- pixman/pixman-utils.c.orig	2008-09-09 08:28:23.000000000 -0400
+++ pixman/pixman-utils.c
@@ -62,6 +62,9 @@ pixman_transform_point_3d (pixman_transf
     return TRUE;
 }
 
+#if defined(USE_SSE2) && defined(__GNUC__) && !defined(__x86_64__) && !defined(__amd64__)
+__attribute__((__force_align_arg_pointer__))
+#endif
 PIXMAN_EXPORT pixman_bool_t
 pixman_blt (uint32_t *src_bits,
 	    uint32_t *dst_bits,
@@ -156,6 +159,9 @@ pixman_fill32 (uint32_t *bits,
     }
 }
 
+#if defined(USE_SSE2) && defined(__GNUC__) && !defined(__x86_64__) && !defined(__amd64__)
+__attribute__((__force_align_arg_pointer__))
+#endif
 PIXMAN_EXPORT pixman_bool_t
 pixman_fill (uint32_t *bits,
 	     int stride,
